src/rendering/SceneManager.cc

```diff
@@ -2336,3 +2253,102 @@ AnimationUpdateData SceneManagerPrivate::ActorTrajectoryAt(
   animData.valid = true;
   return animData;
 }
+
+std::vector<common::TrajectoryInfo>
+SceneManager::LoadTrajectories(const sdf::Actor &_actor,
+  std::unordered_map<std::string, unsigned int> &_mapAnimNameId,
+  Entity &_id)
+{
+
+  rendering::MeshDescriptor descriptor;
+  descriptor.meshName = asFullPath(_actor.SkinFilename(), _actor.FilePath());
+  common::MeshManager *meshManager = common::MeshManager::Instance();
+  descriptor.mesh = meshManager->Load(descriptor.meshName);
+  common::SkeletonPtr meshSkel = descriptor.mesh->MeshSkeleton();
+
+  std::vector<common::TrajectoryInfo> trajectories;
+
+  if (_actor.TrajectoryCount() != 0)
+  {
+    // Load all trajectories specified in sdf
+    for (unsigned i = 0; i < _actor.TrajectoryCount(); i++)
```

Comment:
```suggestion
    for (unsigned i = 0; i < _actor.TrajectoryCount(); ++i)
```

Reply:
Is there a specific reason for that? Is it safer?

Reply:
Done.